### ISSUE TITLE
E701 false positive after migrating from 244 tom 245

### ISSUE THREAD

**Issue by alexprengere** [charliermarsh/ruff#2759]
This used to be fine, but now raises `E701 Multiple statements on one line (colon)`:

```python
cond = False
func = lambda x: x** 2 if cond else lambda x:x
```

**Comment by charliermarsh** [comment]
Thanks! Will fix.
